sdk/python/kfp/dsl/_container_op.py

```diff
@@ -1062,6 +1062,15 @@ def __init__(self, argument, input=None, path=None):
     self.path = path
 
 
+def _is_legacy_output_name(output_name: str) -> Tuple[bool, str]:
+    normalized_output_name = re.sub('[^a-zA-Z0-9]', '-', output_name.lower())
```

Comment:
I think we have this transform in many places, shall we extract a common util?

Reply:
I think that's what I just did here? :-)

Reply:
what's the difference between this and https://github.com/kubeflow/pipelines/blob/e7c281bd65ca79c20c2ae52a4dc4798241333b53/sdk/python/kfp/compiler/_k8s_helper.py#L20?

Reply:
and
* https://github.com/kubeflow/pipelines/blob/cc83e1089b573256e781ed2e4ac90f604129e769/sdk/python/kfp/components/_naming.py#L51-L53
* https://github.com/kubeflow/pipelines/blob/cc83e1089b573256e781ed2e4ac90f604129e769/sdk/python/kfp/dsl/_pipeline_param.py#L25-L41
* https://github.com/kubeflow/pipelines/blob/fdab2e679cad415871b8d864902e1b7e1e292451/sdk/python/kfp/dsl/dsl_utils.py#L51-L58

looks like there's an implementation everywhere...

this is a non blocker for this PR

Reply:
Sorry, I don't see your point here. These different functions are used in different places. I don't think the goal of this PR is to refactor and clean up this code.